### DISCUSSION TITLE
How to use `patches` in a channel flake input in develop?

### DISCUSSION THREAD

**Discussion by andrevmatos** [nrdxp/nixdev#349] [Q&A]
I'm using current `develop` branch, and I'm trying to apply some patches from upstream's open PRs to my current `nixos` system. Trying to add `patches` array directly to `channels.nixos` in `flake.nix` gives me the following error:

```
error: The option `channels.nixos.patches' does not exist. Definition values:
       - In `<unknown-file>':
           [
             /nix/store/lflzsw176h3p295fy7jp7zmxbss9fjnq-source/pkgs/127389.patch
           ]
(use '--show-trace' to show detailed location information)
```

So it seems despite fup accepting the `patches` input, some module type declaration doesn't like it. How can I use fup's channel `patches` functionality on current devos?

It'd be interesting to have some documentation on it, specially if there's a way to automatically fetch these patches remotely directly on `flake.nix` body or `flake = false` inputs.

**Comment by Pacman99** [comment]
Yeah to use patches we have to add the option to the channel submodule. We're sort of holding out for https://github.com/NixOS/nix/issues/3920 as that would be the optimal solution. But I would be ok to review a PR to add the option to digga(can't say the same for others) as a temporary solution. Its just not something I plan to do myself, since I don't have much need for patches.

**Reply by blaggacao** [reply]
in addition, `fup`'s patches only work for `nixpkgs`.
